app/views/spree/admin/variants/_form.html.haml

```diff
@@ -62,7 +62,7 @@
           = f.text_field :on_hand
 
 .right.six.columns.omega.label-block
-  - if @product.variant_unit != 'weight'
+  - if @product.variant_unit == 'weight'
```

Comment:
Interesting! Very surprising. I understand that you are hiding this for non-weight variants. But what I don't understand is why it was hidden from weight variants. Was that a mistake? Do we want to show this form for weight variants? What changed?

Reply:
I just checked that page and it looks like we now have two "Weight" inputs:
![Screenshot from 2022-01-25 11-20-14](https://user-images.githubusercontent.com/3524483/150887182-0228aacc-1dca-4c3a-85ef-cad8120a69e4.png)

The existing one is `unit_value_human` and the one you are adding is `variant[weight]`. I haven't looked into the code but I would guess that unit_value_human is translated into the variant's weight attribute. In that case, we can simply get rid of this little part of the form. Is that right?

Reply:
It looks like our specs could be better. This line is checking for the existing weight input:
https://github.com/openfoodfoundation/openfoodnetwork/blob/662bc6816dbc73f0727da395244ef87ecd528dbe/spec/system/admin/variants_spec.rb#L119

But since it's checking by name, it doesn't see that there are two "WEIGHT (KG)" fields now. We should change the spec to check for the label to make it more reliable.

It also looks like there's some obsolete Spree code in that spec.